### PR TITLE
Change from customized paranoid model to sequelize paranoid mode

### PR DESCRIPTION
**Pull request by UncleBu1728**
changed `Group`, `User` and `GroupUser` from using the customized `ParanoidModel` to the sequelize `paranoid` setting

### CODE REVIEW

File: src/api/database/models/Group.ts [hunk 1]
```diff
@@ -25,12 +32,18 @@ import Role from "shared/Role";
  * A group is said to be "owned" if the partnership or interview field is non-null.
  * Otherwise the group is said to be "unowned".
  */
-@Table({ tableName: "groups", modelName: "group" })
+@Table({ paranoid: true, tableName: "groups", modelName: "group" })
 @Fix
-class Group extends ParanoidModel<
+class Group extends Model<
   InferAttributes<Group>,
   InferCreationAttributes<Group>
```
Comment (weihanwang) [inline src/api/database/models/Group.ts:39]: remove these two lines from all models

File: src/api/database/models/Group.ts [hunk 2]
```diff
@@ -25,12 +32,18 @@ import Role from "shared/Role";
...
   > {
```
Comment (weihanwang) [inline src/api/database/models/Group.ts:40]: Remove NonAttribute<> from everywhere

Comment (weihanwang) [reply src/api/database/models/Group.ts]: I don't see this comment was addressed.

**Comment by UncleBu1728** [comment]
> changed `Group`, `User` and `GroupUser` from using the customized `ParanoidModel` to the sequelize `paranoid` setting

Removed `InferAttributes` in the previous commit
Removed `Nonattribute` branding in the latest commit 

The following tests are completed and passed
- Syncing database with an empty db √
- Inside the db tables
  - Adding and deleting rows √
  - Manipulating existing data √
  - Deleting data in the paranoid model √
  - Adding columns √
    - Do not drop columns with syncing
- Passed both on Mac and Windows systems

TODO: 
-fixing yarn `gen-test-data`

**Review by weihanwang — APPROVED** [review]

**Pull request merged**